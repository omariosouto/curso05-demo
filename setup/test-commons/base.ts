import type {Config} from '@jest/types';
import path from 'path'; 

// Sync object
const config: Config.InitialOptions = {
  testEnvironment: 'jsdom',
  setupFilesAfterEnv: [
    path.resolve(__dirname, 'src', 'setupTests.js'),
  ],
  verbose: true,
  transform: {
    '\\.[jt]sx?$': ['babel-jest', {
      configFile: path.resolve(__dirname, '.babelrc'),
    }],
  }
};
export default config;
