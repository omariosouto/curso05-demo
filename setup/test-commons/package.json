{
  "name": "@alura/test-commons",
  "version": "0.0.0",
  "private": true,
  "main": "index.js",
  "files": [
    "base.ts"
  ],
  "dependencies": {
    "jest": "27.0.0",
    "ts-jest": "^27.1.4",
    "ts-node": "^10.7.0"
  }
}
