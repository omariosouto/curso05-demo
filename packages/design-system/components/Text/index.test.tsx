import { customRender } from "@alura/test-commons/react-testing-library";
import { Text } from "./index";

const render = customRender();

describe("expect home", () => {
  it("expect home", () => {
    const { container } = render(<Text tag="h1">Sample text</Text>);
    expect(container).toMatchSnapshot();
  });
});
