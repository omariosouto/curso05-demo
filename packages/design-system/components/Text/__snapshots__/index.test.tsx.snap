// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect home expect home 1`] = `
.c0 {
  font-family: sans-serif;
}

<div>
  <h1
    class="c0"
  >
    Sample text
  </h1>
</div>
`;
